src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/SparkSessionTests.cs

```diff
@@ -65,5 +68,95 @@ public void TestSignaturesV2_4_X()
         {
             Assert.IsType<SparkSession>(SparkSession.Active());
         }
+
+        /// <summary>
+        /// Test CreateDataFrame APIs.
+        /// </summary>        
+        [Fact]
+        public void TestCreateDataFrame()
+        {            
+            // Calling CreateDataFrame with schema
+            {
+                var data = new List<GenericRow>();
+                data.Add(new GenericRow(new object[] { "Alice", 20 }));
+                data.Add(new GenericRow(new object[] { "Bob", 30 }));
+
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("Name", new StringType()),
+                    new StructField("Age", new IntegerType())
+                });
+                DataFrame df = _spark.CreateDataFrame(data, schema);
+                ValidateDataFrame(data.Select(a => a.Values), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<string> _) without schema
+            {
+                var data = new List<string>(new string[] { "Alice", "Bob" });
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("_1", new StringType())
+                });
+
+                DataFrame df = _spark.CreateDataFrame(data);                
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<int> _) without schema
+            {
+                var data = new List<int>(new int[] { 1, 2 });
+                var schema = new StructType(new List<StructField>()
```

Comment:
Can we copy `SchemaWithSingleColumn` here and reuse it?

Reply:
I will have to make `SchemaWithSingleColumn` `internal` then instead of `private`?

Reply:
I mentioned you can "copy" the function here.

Reply:
Oops sorry I misread!